src/hooks/useEditVideoSheet.tsx

```diff
@@ -74,27 +159,74 @@ const useEditVideoSheetContext = () => {
 
 export type EditVideoSheetState = 'closed' | 'open' | 'minimized'
 export const useEditVideoSheet = () => {
+  return useEditVideoSheetContext()
+}
+
+export type EditVideoFormFields = {
+  title: string
+  description: string
+  language: string | null
+  category: string | null
+  hasMarketing: boolean | null
+  isPublic: boolean
+  isExplicit: boolean | null
+  publishedBeforeJoystream: string | null
+  assets: InputFilesState
+}
+
+export const useEditVideoSheetTabData = (tab?: EditVideoSheetTab) => {
   const {
-    videoTabs,
-    addVideoTab,
-    removeVideoTab,
-    selectedVideoTab,
-    resetVideoTabs,
-    setSelectedVideoTab,
-    sheetState,
-    setSheetState,
-  } = useEditVideoSheetContext()
+    activeUser: { channelId },
+  } = useActiveUser()
+  const { drafts } = useDrafts('video', channelId ?? '')
+
+  const { selectedVideoTabCachedAssets } = useEditVideoSheet()
+
+  const { video, loading, error } = useVideo(tab?.id ?? '', { skip: tab?.isDraft })
+
+  if (!tab) {
+    return {
+      data: null,
+      loading: false,
+      error: null,
+    }
+  }
+
+  const draft = drafts.find((d) => d.id === tab.id)
+
+  const assets: InputFilesState = tab.isDraft
+    ? selectedVideoTabCachedAssets || { video: null, thumbnail: null }
+    : {
+        video: {
+          url: createUrlFromAsset(video?.mediaAvailability, video?.mediaUrls, video?.mediaDataObject),
+        },
+        thumbnail: {
+          url: createUrlFromAsset(
+            video?.thumbnailPhotoAvailability,
+            video?.thumbnailPhotoUrls,
+            video?.thumbnailPhotoDataObject
+          ),
+        },
+      }
+
+  const normalizedData: EditVideoFormFields = {
+    title: tab.isDraft ? draft?.title ?? 'New Draft' : video?.title ?? '',
+    description: (tab.isDraft ? draft?.description : video?.description) ?? '',
+    category: (tab.isDraft ? draft?.category : video?.category?.id) ?? null,
+    language: (tab.isDraft ? draft?.language : video?.language?.iso) ?? 'en',
+    isPublic: (tab.isDraft ? draft?.isPublic : video?.isPublic) ?? true,
+    isExplicit: (tab.isDraft ? draft?.isExplicit : video?.isExplicit) ?? null,
+    hasMarketing: (tab.isDraft ? draft?.hasMarketing : video?.hasMarketing) ?? false,
+    publishedBeforeJoystream:
+      (tab.isDraft ? draft?.publishedBeforeJoystream : video?.publishedBeforeJoystream?.toISOString()) ?? null,
+    assets,
```

Comment:
got the error, after clicking one of the videos.
![image](https://user-images.githubusercontent.com/51168865/116682500-9e4a3880-a9ae-11eb-8662-4782ab5d8732.png)
 Not sure if you can reproduce. Maybe this particular video was broken somehow, or this is because the processor is down right now

Reply:
Oh, I think it was the video. The id of the video is  "72". It has set up `publishedBeforeJoystream` field

Reply:
Fixed